cloudferry_devlab/Vagrantfile

```diff
@@ -1,209 +1,77 @@
-      end
-
-      thisnode.vm.box = nodedata['box']
-      thisnode.vm.hostname = nodedata.fetch("hostname", nodename)
-      thisnode.vm.provision "shell", inline: "echo '#{etc_hosts}' >> /etc/hosts"
-      thisnode.vm.provision "shell",
-        path: "./provision/keys.sh",
-        args: [ "--public-key", File.read("#{ENV["HOME"]}/#{options['public_key_path']}").strip() ]
-
-      case nodedata.fetch("role", "")
-        when "openstack"
-          thisnode.vm.provision "shell", path: "./provision/configure_networking.sh", args: ['eth1', 'br-ex', nodedata['ip']]
-          thisnode.vm.provision "shell", path: "./provision/configure_networking.sh", args: ['eth2', 'br-ex2', nodedata['ip2']]
-          thisnode.vm.provision "shell", path: "./provision/configure_openstack.sh", args: [nodedata['ip'], nodename]
-          thisnode.vm.provision "shell", path: "./provision/create_disks_swift.sh"
-
-          if nodename == "icehouse" then
-            thisnode.vm.provision "shell", path: "./provision/cleanup_nova_instances.sh"
-          end
-
-          old_nfs_name = nodedata.fetch("nfs_share_name")
-          if new_nfs_name and new_nfs_name != old_nfs_name
-          then
-            thisnode.vm.provision "shell",
-              path: "./provision/configure_nfs.sh",
-              args: [old_nfs_name, new_nfs_name]
-          end
-        when "lab"
-          config.vm.provision "shell", path: "./provision/prerequisites.sh"
-          thisnode.vm.provision "shell",
-            path: "./provision/cloudferry.sh"
-          if File.exist?(File.join(Dir.home, ".ssh/id_rsa")) then
-            thisnode.vm.provision "file",
-              source: "~/.ssh/id_rsa",
-              destination: "/home/vagrant/.ssh/id_rsa"
-          end
-      end
-
-      thisnode.vm.network "private_network", ip: nodedata['ip'], auto_config: (nodedata.fetch("role", "") != "openstack"), nic_type: "virtio"
-      thisnode.vm.network "private_network", ip: nodedata['ip2'], auto_config: (nodedata.fetch("role", "") != "openstack"), nic_type: "virtio"
-
-      thisnode.vm.provider "virtualbox" do |v|
-        v.memory = nodedata.fetch("memory", 1024)
-        v.cpus = nodedata.fetch("cpus", 2)
-        v.customize ["modifyvm", :id, "--nicpromisc2", "allow-all"]
-        v.customize ["modifyvm", :id, "--nicpromisc3", "allow-all"]
-        v.customize ["modifyvm", :id, "--cpuexecutioncap", "90"]
-        v.customize ["modifyvm", :id, "--nictype1", "virtio"]
+      role = nodedata[:role]
+      thisnode.vm.box_url = nodedata[:url]
+
+      thisnode.vm.box = nodedata[:box]
+      thisnode.vm.hostname = nodename.to_s
+      thisnode.vm.provision 'shell', inline: "echo '#{etc_hosts}' >> /etc/hosts"
+      thisnode.vm.provision 'shell',
+                            path: './provision/add_migration_user.sh',
+                            args: [migrate_user, password]
+      thisnode.vm.provision 'shell',
+                            path: './provision/keys.sh',
```

Comment:
`./provision/keys.sh` should also provision private key, otherwise there's no way to copy data over ssh from grizzly to icehouse box, for example. In the end following connectivity should be available for the end-user:

# ```developer@workstation $ ssh migration@grizzly -i private_key```
# ```developer@workstation $ ssh migration@icehouse -i private_key```
# ```developer@workstation $ ssh migration@juno -i private_key```
# ```developer@workstation $ ssh migration@nfs -i private_key```
# ```migration@grizzly $ ssh migration@icehouse -i ~/.ssh/private_key```
# ```migration@grizzly $ ssh migration@juno -i ~/.ssh/private_key```
# ```migration@grizzly $ ssh migration@nfs -i ~/.ssh/private_key```
# ```migration@icehouse $ ssh migration@grizzly -i ~/.ssh/private_key```
# ```migration@icehouse $ ssh migration@juno -i ~/.ssh/private_key```
# ```migration@icehouse $ ssh migration@nfs -i ~/.ssh/private_key```
# ```migration@juno $ ssh migration@grizzly -i ~/.ssh/private_key```
# ```migration@juno $ ssh migration@icehouse -i ~/.ssh/private_key```
# ```migration@juno $ ssh migration@nfs -i ~/.ssh/private_key```